lang/cs/Org.Apache.REEF.IO.Tests/TestAzureDataLakeFileSystemE2E.cs

```diff
@@ -0,0 +1,229 @@
+            Assert.False(_fileSystem.Exists(PathToFile(fileName)));
+        }
+
+        [Fact(Skip = SkipMessage)]
+        public void TestCopyE2E()
+        {
+            throw new NotImplementedException();
+        }
+
+        [Fact(Skip = SkipMessage)]
+        public void TestCopyToLocalE2E()
+        {         
+            var tempFilePath = Path.GetTempFileName();
+            try
+            {
+                string fileName = UploadFromString(ContentsText);
+                _fileSystem.CopyToLocal(PathToFile(fileName), tempFilePath);
+                Assert.True(File.Exists(tempFilePath));
+                Assert.Equal(ContentsText, File.ReadAllText(tempFilePath));
+            }
+            finally
+            {
+                File.Delete(tempFilePath);
+            }
+        }
+
+        [Fact(Skip = SkipMessage)]
+        public void TestCopyFromLocalE2E()
+        {
+            var tempFilePath = Path.GetTempFileName();
+            var tempFileName = Path.GetFileName(tempFilePath);
+            try
+            {
+                File.WriteAllText(tempFilePath, ContentsText);
+                Uri remoteFileUri = PathToFile($"/{defaultFolderName}/{tempFileName}");
+                _fileSystem.CopyFromLocal(tempFilePath, remoteFileUri);
+                Assert.True(_adlsClient.CheckExists($"/{defaultFolderName}/{tempFileName}"));
+                var stream = _fileSystem.Open(remoteFileUri);
+                StreamReader reader = new StreamReader(stream);
+                string streamText = reader.ReadToEnd();
+                Assert.Equal(ContentsText, streamText);
+            }
+            finally
+            {
+                File.Delete(tempFilePath);
+            }
+        }
+
+        [Fact(Skip = SkipMessage)]
+        public void TestCreateDirectoryE2E()
+        {
+            string dirName = $"/{defaultFolderName}";
+            _fileSystem.CreateDirectory(PathToFile(dirName));
+            Assert.True(_adlsClient.CheckExists(dirName));
+        }
+
+        [Fact(Skip = SkipMessage)]
+        public void TestDeleteDirectoryE2E()
+        {
+            string dirName = $"/{defaultFolderName}";
+            _adlsClient.CreateDirectory(dirName);
+            Assert.True(_adlsClient.CheckExists(dirName));
+            _fileSystem.Delete(PathToFile(dirName));
+            Assert.False(_adlsClient.CheckExists(dirName));
+        }
+
+        [Fact(Skip = SkipMessage)]
+        public void TestGetChildrenE2E()
+        {
+            string fileName1 = UploadFromString("file1", 1);
+            string fileName2 = UploadFromString("file2", 2);
+            string dirName = $"/{defaultFolderName}";
+            var childUris = _fileSystem.GetChildren(PathToFile(dirName)).ToList();
+            Assert.True(childUris.Count() == 2);
+            Assert.Equal(fileName2,  childUris.First().AbsolutePath.Equals(fileName1) ? childUris.Last().AbsolutePath : childUris.First().AbsolutePath);
```

Comment:
This is overly complex. Please just use a collection assert
```csharp
CollectionAssert.AreEquivalent(new List [] { fileName1, fileName2}, childUris);
```

Reply:
fixed